docs/docs/guides/build/assets/asset-selection-syntax/reference.md

```diff
@@ -0,0 +1,95 @@
+- **Downstream assets** receive input from a given asset.
+- A **layer** is a level of connection in the graph. One layer up or down means a direct connection, "two layers" means connections two steps away, and so on.
+
+| Asset selection | Syntax | Description |
+|------|--------|-------------|
+| **All layers upstream of an asset** | `+key:"my_asset"` | Selects all upstream assets that provide input into `my_asset`. |
+| **One layer upstream of an asset** | `1+key:"my_asset"` | Selects assets that directly provide input to `my_asset`. |
+| **Two layers upstream of an asset** | `2+key:"my_asset"` | Selects assets two steps upstream from `my_asset`. |
+| **All layers downstream of an asset** | `key:"my_asset"+` | Selects all downstream assets that depend on `my_asset`. |
+| **One layer downstream of an asset** | `key:"my_asset"+1` | Selects assets that directly receive input from `my_asset`. |
+| **Two layers downstream of an asset** | `key:"my_asset"+2` | Selects assets two steps downstream from `my_asset`. |
+| **One layer upstream and downstream of an asset** | `1+key:"my_asset"+1` | Selects one layer of assets providing input to and receiving input from `my_asset`. |
+| **Two layers upstream and downstream of an asset** | `2+key:"my_asset"+2` | Selects two layers of assets providing input to and receiving input from `my_asset`. |
+| **All layers upstream and downstream of an asset** | `+key:"my_asset"+` | Selects all assets upstream and downstream of `my_asset`. |
+
+## Operands and grouping
+
+You can combine multiple filters with operands and group them with parentheses to further refine your asset selection:
+
+| Operand | Syntax | Description |
+|---------|--------|-------------|
+| **`and`** | `owner:"alice" and kind:"dbt"` | Selects assets owned by `alice` of kind `dbt`. |
+| **`or`** | `owner:"billing" or owner:"sales"` | Selects assets owned by either `billing` or `sales`. |
+| **`not`** | `not tag:"obsolete"` | Excludes assets tagged with `obsolete`. |
+| **Grouping `()`** | `(owner:"alice" and kind:"table") or group:"analytics"` | Symbols `(` and `)` used to group expressions and control the order of evaluation in queries. This example selects assets that are both owned by `alice` and of kind `table`, or that belong to the `analytics` group. |
+
+## Functions
+
+You can use `sinks()` and `roots()` functions to return the sink and root assets of an asset selection.
+
+- **Sink assets** are assets without any downstream dependencies (leaf nodes), which means they don't provide input to any other assets.
+- **Root assets** are assets without any upstream dependencies (root nodes), which means no assets provide input to them.
+
+| Function | Syntax | Description |
+|--------|-------------|-----------|
+| **`sinks(expr)`** | `sinks(owner:"team:billing")` | Find all sink assets depending on assets by the billing team. |
+| **`roots(expr)`** | `roots(owner:"team:billing")` | Find all root assets depended on by the billing team. |
```

Comment:
From dogfooding on docs + feature people were a bit confused by what the functions did.

This is the feedback: https://www.notion.so/dagster/Selection-Syntax-Dogfooding-1a718b92e4628060a3bee29204548734?pvs=4#1ac18b92e462809a9cfbe8cd77516141.

We need to make it clear we're find the Sinks/Roots OF the expression, and not WITHIN the expression.

Reply:
Gotcha, that makes sense -- is this accurate?

| Example | Description |
|--------|--------------|
| `sinks(owner:"team:billing")` | Find all assets without any downstream dependencies that depend on assets owned by the billing team. |
| `roots(owner:"team:billing")` | Find all assets without any upstream dependencies that feed into assets owned by the billing team. |

(I'll also add `sinks(*)` and roots(*)` to that table.)

Reply:
I'll also add a note specifying that we're finding the sinks/roots of the expression, not within the expression.